src/api/utils.ts

```diff
@@ -9,21 +9,26 @@ const HttpError = z.object({
 });
 type HttpError = z.infer<typeof HttpError>;
 
-/** https://github.com/AtB-AS/amp-rs/blob/main/amp-http/src/lib.rs */
 export const ErrorResponse = z.object({
-  http: HttpError,
   kind: z.string(),
   message: z.string().nullish(),
   details: z.array(z.unknown()).nullish(),
 });
 export type ErrorResponse = z.infer<typeof ErrorResponse>;
+/** https://github.com/AtB-AS/amp-rs/blob/main/amp-http/src/lib.rs */
+export const HttpErrorResponse = ErrorResponse.extend({
+  http: HttpError,
+});
```

Comment:
I still think it would be nice if we could keep one of these types a 1:1 with the backend type. 

One solution could be to keep `ErrorResponse` on the same format as before, then maybe the name `RequestError` would make sense, which has the `http` field overwritten as undefined.

Not a blocker though. I'm ok with the current solution as well, and open for other suggestions 🤷

Reply:
I like the idea of 1:1 as well, I can look at your suggestion. That would affect the comments below as well, the casting to `Http...`.